igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/model/GenericArtifact.groovy

```diff
@@ -16,14 +16,49 @@
 
 package com.netflix.spinnaker.igor.build.model
 
+import com.fasterxml.jackson.annotation.JsonInclude
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
 class GenericArtifact {
     String fileName
     String displayPath
     String relativePath
+    String reference
+    String name
+    String type
+    String version
 
     GenericArtifact(String fileName, String displayPath, String relativePath) {
         this.fileName = fileName
         this.displayPath = displayPath
         this.relativePath = relativePath
     }
+
+    String toString() {
```

Comment:
This will not produce the output you want:
`fileName:,somefilename.jardisplayPath:,somedisplaypathrelativePath:,somerelativepathreference:,This,is,a,referencename:,Name,of,artifacttype:,jarversino:,123`
You could instead just annotate the class with `@ToString(includeNames=true)` and remove this method.